pkg/sync/expand.go

```diff
@@ -200,10 +200,13 @@ func (d *EntitlementGraph) AddEdge(srcEntitlementID string, dstEntitlementID str
 	if !ok {
 		d.Edges[srcEntitlementID] = make(map[string]*edgeInfo)
 	}
-	d.Edges[srcEntitlementID][dstEntitlementID] = &edgeInfo{
-		Expanded:        false,
-		Shallow:         shallow,
-		ResourceTypeIDs: resourceTypeIDs,
+	_, ok = d.Edges[srcEntitlementID][dstEntitlementID]
+	if !ok {
+		d.Edges[srcEntitlementID][dstEntitlementID] = &edgeInfo{
+			Expanded:        false,
+			Shallow:         shallow,
+			ResourceTypeIDs: resourceTypeIDs,
+		}
```

Comment:
When testing, I did add an else with a panic(), but we never hit it. I'm not sure if I should keep that in the code or not. I think if we try to add an existing edge, something has gone horribly wrong.